### ISSUE TITLE
a11y: Fix keyboard navigation, labels/descriptions, roles and focus display for miscellaneous single occurrences

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12811]
a11y issues: 223, 177, 166, 155, 145, 179

**Issue 223 Import Extension Catalog -  Keyboard focus traps**

Keyboard only users are not able to tab pass when the users navigate the page using 'Tab' key and focus gets trapped on combobox which resulted in keyboard-only users not being able to access the interactive elements after the form fields. Additionally, when users try to come out from mentioned combobox modal dialog gets closed.

![image](https://github.com/user-attachments/assets/3a6ac4bc-048c-466c-8289-788e6bc4c8a3)

Suggested remediation:

Ensure that the keyboard focus does not get trapped on any component and users can navigate the page content with ease in all the browsers. Modify the scripts associated with the dropdown list to ensure that keyboard trap can be avoided.

**Issue 177 Pod Security Admission: Creates - Role button not defined**

Role is not defined for the interactive image, making it difficult for screen reader users to access the functionality associated with it - issue is with the close button.

![image](https://github.com/user-attachments/assets/8f76f357-72cb-446f-bc74-e30f4752d996)

Suggested remediation:

Ensure that the interactive images is programmatically identified as buttons.
Apply the following changes:
• Code the mentioned interactive image using <button> element. Alternatively, use the role="button" and tabindex="0" attributes to define the button role.
• Modify the scripts to ensure that the image button is operable with keyboard and a mouse.
• Provide textual description such as "Close" for image button using "aria-label" attribute.

**Issue 166 Charts - Tab not programmatically determined**

The mentioned interactive elements are not programmatically determined as tab. Also, the number of tabs are not programmatically determined. On activating the interactive elements, the related content gets dynamically updated below. Although the dynamic change in the content is easier to understand for sighted users, it is not intuitive for screen reader users.

![image](https://github.com/user-attachments/assets/a718a990-38f2-4977-b848-6ba49fe547c5)

Suggested remediation:

Apply the following changes:
• Set the role="tablist" for the <div> tag containing the elements that behaves as a tab.
• Set the role="tab" for the <button> tags used for displaying "Browse" and "Featured". In addition, specify the ARIA attributes for tabs, such as "aria-controls" and "aria-selected". (The value of the aria-selected attribute needs to change on the basis of currently active tab.)
o Set aria-selected="true" for the active tab.
o Specify an "id" attribute for the <div> tags used to display the content of respective tab. Reference the value of the "id" attribute correctly via "aria-controls" attribute of each tab.
o Specify an 'id' attribute for the <button> tags used to display each tab. Reference the value of the "id" attribute correctly via "aria-labelledby" attribute of each tab panel.
• Set the role="tabpanel" for the <div> tags displaying the contents associated with "Browse" and "Featured" tabs.
o As best practice, provide tabindex="0" attribute on the container of tab panel.
• Ensure that the keyboard focus cycles between the tabs using arrow keys. The not selected tab should not receive the keyboard focus when navigated using "Tab" key.

More information can be found at https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-manual.html

**Issue 155 Delete Project - Missing form label or 'title' attribute**

The form field do not have the label associated with them resulting in screen reader users fail to understand the purpose of the form field.

![image](https://github.com/user-attachments/assets/e275e153-939a-4c84-88ae-2670f17a7e27)

Suggested remediation:

Apply the following changes:
• Replace the <div> tag containing the "Enter default below to confirm" text using <label> tag.
• Use "for" and "id" attributes to associate label with their respective input field. Ensure that the value of "for" attribute of the label exactly matches the value of "id" attribute of the input field.

**Issue 145 Kubectl Shell - Role and state not defined for interactive images**

The role and state are not programmatically defined for the interactive image making it difficult for screen reader users to access the state information and functionality associated with interactive image.

![image](https://github.com/user-attachments/assets/35cc2b49-e417-4f97-8ac8-4bc0d9b4ae25)

Suggested remediation:

Apply the following changes:
• Code the mentioned interactive elements using <button> tag. Alternatively, use role="button" and tabindex="0" attributes to define the button role.
• Modify the scripts to ensure that the functionality is accessible for all users.
• Set "aria-expanded" attribute to "false" for the image buttons when the buttons are in the collapsed state. Ensure that the value of "aria-expanded" attribute changes on user interaction via scripting.
• Provide textual description as "Resize" for the image button using "title" or "aria-label" attribute.

**Issue 179 Pod Security Admission: Create - Missing form label or 'title' attribute**

The form fields do not have the label associated with them and "title" attribute is not included in their source code either.
As a result, screen reader users will not understand the purpose of the form fields.

Pod Security Admission: Create
- Version (default: latest)
- Enter a comma separated list of usernames 
- Enter a comma separated list of runtimeClasses
- Enter a comma separated list of namespaces

![image](https://github.com/user-attachments/assets/71462cfa-7bd5-4460-96a9-36805ae21e0b)

Suggested remediation:

Apply the following changes:
• Provide unique and descriptive visual labels such as "Enforce - Version (default: latest)", "Audit - Version (default: latest)", "Usernames - Enter a comma separated list of usernames", "Namespaces - Enter a comma separated list of namespaces" and so on for the mentioned form fields using <label> tag.
• Use "for" and "id" attributes to associate labels with their respective form fields. Ensure that the value of "for" attribute of the label exactly matches the value of "id" attribute of the form field.

**Comment by izaac** [comment]
Rancher version: `v2.11-41098eee7331faf5d4a478dfcf66232867ed6a6f-head`
UI Version: `master 95c0062`
Firefox version: `135.0.1`
Chrome version: `133.0.6943.141`

This is related to the validation done in https://github.com/rancher/dashboard/issues/12795

Comboxes no longer trap navigation focus.  And selecting options works using the keyboard as expected. ✅
